app/presenters/tree_builder_ops_rbac_features.rb

```diff
@@ -0,0 +1,160 @@
+class TreeBuilderOpsRbacFeatures < TreeBuilder
+  has_kids_for Menu::Section,     [:x_get_tree_section_kids]
+  has_kids_for Menu::Item,        [:x_get_tree_item_kids]
+  has_kids_for MiqProductFeature, [:x_get_tree_feature_kids]
+
+  attr_reader :role, :features, :editable, :root_counter
+
+  def initialize(name, type, sandbox, build, role:, editable: false)
+    @role     = role
+    @editable = editable
+    @features = @role.miq_product_features.order(:identifier).pluck(:identifier)
+
+    @root_counter = []
+
+    # Make sure tree_state doesn't hold on to old data between requests
+    TreeState.new(sandbox).remove_tree(name)
```

Comment:
I don't like it either, but otherwise between certain requests (like going to edit screen) the tree will come from a cached version. This is done in another TreeBuilder subclass too.